toolchain/sem_ir/typed_insts.h

```diff
@@ -102,8 +102,7 @@ struct AnyAggregateAccess {
 // Common representation for aggregate index nodes, which access an element
 // determined by evaluating an expression.
 struct AnyAggregateIndex {
-  static constexpr InstKind Kinds[] = {InstKind::ArrayIndex,
-                                       InstKind::TupleIndex};
+  static constexpr InstKind Kinds[] = {InstKind::ArrayIndex};
```

Comment:
If only an `ArrayIndex` can be an `AnyAggregateIndex`we should be able to drop `AnyAggregateIndex` and use `ArrayIndex` instead.